// oneai/lib/src/pipeline.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var pipeline_1 = require("./model/pipeline");
var schedule_1 = require("./schedule");
var _Pipeline = /** @class */ (function () {
    function _Pipeline() {
        var steps = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            steps[_i] = arguments[_i];
        }
        this.steps = steps;
    }
    _Pipeline.prototype.run = function (text, params) {
        return __awaiter(this, void 0, void 0, function () {
            var input, response;
            var _this = this;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        input = (0, pipeline_1.wrapContent)(text);
                        return [4 /*yield*/, this.client.postPipeline(input, this.steps, params)];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/, (0, pipeline_1.isInput)(response) || (params === null || params === void 0 ? void 0 : params.polling) === false ? response
                                : (0, schedule_1.polling)(response, function (t) { return _this.client.getTaskStatus.call(_this.client, t); }, (params === null || params === void 0 ? void 0 : params.interval) || 1, this.client.logger)];
                }
            });
        });
    };
    _Pipeline.prototype.runFile = function (filePath, params) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.run({ filePath: filePath }, params)];
            });
        });
    };
    _Pipeline.prototype.runBatch = function (inputs, params) {
        return __awaiter(this, void 0, void 0, function () {
            var _this = this;
            return __generator(this, function (_a) {
                return [2 /*return*/, (0, schedule_1.batchProcessing)(inputs, function (input) { return _this.run(input, params); }, (params === null || params === void 0 ? void 0 : params.maxConcurrentRequests) || 2, params === null || params === void 0 ? void 0 : params.onOutput, params === null || params === void 0 ? void 0 : params.onError, this.client.logger)];
            });
        });
    };
    _Pipeline.prototype.taskStatus = function (task) {
        return __awaiter(this, void 0, void 0, function () {
            var taskObj;
            return __generator(this, function (_a) {
                taskObj = (typeof task === 'string') ? { requestId: task, name: '', skills: this.steps } : task;
                return [2 /*return*/, this.client.getTaskStatus(taskObj)];
            });
        });
    };
    _Pipeline.prototype.awaitTask = function (task, params) {
        return __awaiter(this, void 0, void 0, function () {
            var taskObj;
            var _this = this;
            return __generator(this, function (_a) {
                taskObj = (typeof task === 'string') ? { requestId: task, name: '', skills: this.steps } : task;
                return [2 /*return*/, (0, schedule_1.polling)(taskObj, function (t) { return _this.client.getTaskStatus.call(_this.client, t); }, (params === null || params === void 0 ? void 0 : params.interval) || 1, this.client.logger)];
            });
        });
    };
    return _Pipeline;
}());
var createPipelineClass = function (client) { return /** @class */ (function (_super) {
    __extends(class_1, _super);
    function class_1() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.client = client;
        return _this;
    }
    return class_1;
}(_Pipeline)); };
exports.default = createPipelineClass;
