// oneai/lib/src/skills.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.skills = void 0;
exports.skills = {
    summarize: function (params) {
        var skill = ({
            apiName: 'summarize',
            params: params,
            textField: 'summary',
        });
        if ((params === null || params === void 0 ? void 0 : params.find_origins) !== false) {
            skill.params = __assign({}, params);
            skill.labelsField = 'origins';
        }
        return skill;
    },
    emotions: function (params) { return ({ apiName: 'emotions', params: params }); },
    keywords: function (params) { return ({ apiName: 'keywords', params: params }); },
    /** @deprecated since v. 0.0.9- use `splitBySentence` instead */
    sentences: function (params) { return ({ apiName: 'sentences', params: params }); },
    highlights: function (params) { return ({ apiName: 'highlights', params: params }); },
    topics: function (params) { return ({
        apiName: 'article-topics', labelsField: 'topics',
        params: params,
    }); },
    sentiments: function (params) { return ({ apiName: 'sentiments', params: params }); },
    htmlToArticle: function (params) { return ({
        apiName: 'html-extract-article',
        textField: 'htmlArticle',
        labelsField: 'htmlFields',
        params: params,
    }); },
    htmlAllText: function (params) { return ({
        apiName: 'html-extract-text',
        textField: 'htmlText',
        labelsField: 'htmlFields',
        params: params,
    }); },
    proofread: function (params) { return ({
        apiName: 'enhance',
        textField: 'proofread',
        labelsField: 'replacements',
        params: params,
    }); },
    actionItems: function (params) { return ({
        apiName: 'action-items', labelsField: 'actionItems',
        params: params,
    }); },
    anonymize: function (params) { return ({
        apiName: 'anonymize',
        labelsField: 'anonymizations',
        params: params,
    }); },
    names: function (params) { return ({ apiName: 'names', params: params }); },
    numbers: function (params) { return ({ apiName: 'numbers', params: params }); },
    splitBySentence: function (params) { return ({ apiName: 'sentences', params: params }); },
    splitByTopic: function (params) { return ({
        apiName: 'dialogue-segmentation', labelsField: 'segments',
        params: params,
    }); },
    salesInsights: function (params) { return ({
        apiName: 'sales-insights', labelsField: 'salesInsights',
        params: params,
    }); },
    transcribe: function (params) { return ({
        apiName: 'transcribe',
        textField: 'transcription',
        labelsField: 'words',
        params: params,
    }); },
    detectLanguage: function (params) { return ({
        apiName: 'detect-language', labelsField: 'language',
        params: params,
    }); },
    headline: function (params) { return ({
        apiName: 'headline', labelsField: 'headline',
        params: params,
    }); },
    subheading: function (params) { return ({
        apiName: 'subheading', labelsField: 'subheading',
        params: params,
    }); },
    pricing: function (params) { return ({
        apiName: 'business-entities', labelsField: 'pricing',
        params: params,
    }); },
    emailInsights: function (params) { return ({
        apiName: 'service-email-insights', labelsField: 'emailInsights',
        params: params,
    }); },
    clustering: function (params) { return ({
        apiName: 'clustering', labelsField: 'clusters',
        params: params,
    }); },
    pdfExtractText: function (params) { return ({
        apiName: 'pdf-extract-text', textField: 'pdfText',
        params: params,
    }); },
};
