// oneai/lib/src/model/pipeline.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.wrapContent = exports.isCSV = exports.isFile = exports.isInput = void 0;
var fs_1 = __importDefault(require("fs"));
var path = __importStar(require("path"));
function isInput(object) {
    return typeof object === 'object' && 'text' in object;
}
exports.isInput = isInput;
function isFile(object) {
    return typeof object === 'object' && 'filePath' in object;
}
exports.isFile = isFile;
function isCSV(object) {
    return typeof object === 'object' && 'columns' in object;
}
exports.isCSV = isCSV;
var extensions = {
    '.json': {
        contentType: 'application/json',
        type: 'conversation',
        isBinary: false,
    },
    '.txt': {
        contentType: 'text/plain',
        type: 'article',
        isBinary: false,
    },
    '.srt': {
        contentType: 'text/plain',
        type: 'conversation',
        isBinary: false,
    },
    '.mp3': {
        contentType: 'audio/mp3',
        type: 'conversation',
        isBinary: true,
    },
    '.mp4': {
        contentType: 'audio/mpeg',
        type: 'conversation',
        isBinary: true,
    },
    '.wav': {
        contentType: 'audio/wav',
        type: 'conversation',
        isBinary: true,
    },
    '.html': {
        contentType: 'text/plain',
        type: 'article',
        isBinary: false,
    },
    '.csv': {
        contentType: 'text/csv',
        type: 'article',
        isBinary: false,
    },
    '.pdf': {
        contentType: 'text/pdf',
        type: 'article',
        isBinary: true,
    },
};
function prepFileInput(input) {
    var ext = path.extname(input.text.filePath);
    if (!(ext in extensions))
        throw new Error("Unsupported file type: ".concat(ext));
    var _a = extensions[ext], contentType = _a.contentType, type = _a.type, isBinary = _a.isBinary;
    return __assign(__assign({ encoding: (isBinary) ? 'base64' : 'utf8', contentType: contentType, type: type }, input), { text: (input.text.buffer !== undefined) ? input.text : __assign(__assign({}, input.text), { buffer: fs_1.default.readFileSync(input.text.filePath) }) });
}
function wrapContent(content) {
    if (typeof content === 'string') {
        return {
            text: content,
            type: 'article',
        };
    }
    if (isInput(content)) {
        if (isFile(content.text))
            return prepFileInput(content);
        return content;
    }
    if (isFile(content))
        return prepFileInput({ text: content });
    return {
        text: content,
        type: 'conversation',
        contentType: 'application/json',
    };
}
exports.wrapContent = wrapContent;
